plugins/grafana-custom-plugins/grafana-dependency-plugin/README.md

```diff
@@ -0,0 +1,146 @@
+# Grafana Service Dependency Graph Plugin
+
+## What is Grafana Service Dependency Graph Plugin?
+
+Grafana Service Dependency Graph Plugin allows users to visualize how pods send
+and receive traffic to and from other pods and services. To help visualize the
+flows, the plugin groups pods by pod label and shows them 'inside' the node they
+are a part of.
+
+```mermaid
+graph LR;
+subgraph worker_node
+worker_pod_1
+worker_pod_2
+end;
+```
+A line pointing from source to destination is also shown, with the amount of
+data transmitted labelling the line.
+```mermaid
+graph LR;
+subgraph worker_node
+worker_pod_1 -- 8.4 MB --> worker_pod_2
+end;
+```
+## Acknowledgements
+
+The Service Dependency Graph Plugin is created using [mermaid-js](https://mermaid-js.github.io/mermaid/#/)
+
+## Data Source
+
+Supported Databases:
+
+- ClickHouse
+
+## Queries Convention
+
+Currently the Service Dependency Graph Plugin is created for restricted uses,
+only for visualizing transmitted data between pods and services. For correct
+loading of data for the Service Dependency Graph Plugin, the query is expected
+to return the following fields, in arbitrary order.
+
+- field 1: sourcePodName value with name or an alias of `sourcePodName`
+- field 2: sourcePodLabels value with name or an alias of `sourcePodLabels`
+- field 3: sourceNodeName value with name or an alias of `sourceNodeName`
+- field 4: destinationPodName value with name or an alias of `destinationPodName`
+- field 5: destinationPodLabels value with name or an alias of `destinationPodLabels`
+- field 6: destinationNodeName value with name or an alias of `destinationNodeName`
+- field 7: destinationServicePortName value with name or an alias of `destinationServicePortName`
+- field 8: octetDeltaCount value with name or an alias of `octetDeltaCount`
+
+Clickhouse query example:
+```sql
+SELECT sourcePodName,
+sourcePodLabels,
+sourceNodeName,
+destinationPodName,
+destinationPodLabels,
+destinationNodeName,
+destinationServicePortName,
+octetDeltaCount
```

Comment:
Two questions on the query:

1. shall we add a time filter on the query? please refer to other dashboards [queries](https://github.com/antrea-io/theia/blob/b6c1c1f181853a7a4edc9f60769970c2fb0729e3/build/charts/theia/provisioning/dashboards/homepage.json#L119) to see how we add it
2. shall we do `sum(octetDeltaCount)` among the selected time range? otherwise it doesn't make much sense to me by showing the standalone octetDeltaCount value

Reply:
1. yeah, I'll add that as well
2. The plugin reads all individual flows from the query, and they are processed afterwards. The octectDeltaCount itself is never shown, only used to calculate the total which is. If sum() was used, it would give a total of all bytes transferred across all flows, which may not be a helpful metric to the viewer.

Reply:
thanks for the explanation. got it.